modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/networkapi/NetworkApiService.scala

```diff
@@ -0,0 +1,62 @@
+package org.tessellation.rosetta.domain.networkapi
+
+import cats.data.EitherT
+import cats.effect.Async
+
+import org.tessellation.BuildInfo
+import org.tessellation.rosetta.domain.NetworkIdentifier
+import org.tessellation.rosetta.domain.error.{LatestSnapshotNotFound, NetworkApiError}
+import org.tessellation.rosetta.domain.networkapi.model.options._
+import org.tessellation.rosetta.domain.networkapi.model.status._
+import org.tessellation.schema.{GlobalSnapshot, SnapshotOrdinal}
+import org.tessellation.security.Hashed
+import org.tessellation.security.hex.Hex
+
+trait NetworkApiService[F[_]] {
+  def list: EitherT[F, NetworkApiError, List[NetworkIdentifier]]
+  def options: EitherT[F, NetworkApiError, NetworkApiOptions]
+  def status: EitherT[F, NetworkApiError, NetworkStatusResponse]
+}
+
+object NetworkApiService {
+  private val rosettaVersion = "1.4.14" // TODO get from a config?
+  private val version = Version(rosettaVersion = rosettaVersion, nodeVersion = BuildInfo.version)
+
+  def make[F[_]: Async](
+    getLastSnapshot: () => F[Option[Hashed[GlobalSnapshot]]],
+    getGenesisSnapshot: () => F[GlobalSnapshot],
+    getTimestampForSnapshot: SnapshotOrdinal => F[Long]
+  ): NetworkApiService[F] = new NetworkApiService[F] {
+    def list: EitherT[F, NetworkApiError, List[NetworkIdentifier]] =
+      EitherT.pure(List(NetworkIdentifier.Mainnet, NetworkIdentifier.Testnet))
+
+    def options: EitherT[F, NetworkApiError, NetworkApiOptions] =
+      EitherT.pure(NetworkApiOptions(version, Allow.default))
+
+    def status: EitherT[F, NetworkApiError, NetworkStatusResponse] = for {
+      lastSnapshot <- EitherT.fromOptionF[F, NetworkApiError, Hashed[GlobalSnapshot]](
+        getLastSnapshot(),
+        LatestSnapshotNotFound
+      )
+      genesisSnapshot <- EitherT.liftF(getGenesisSnapshot())
+      blockTimestamp <- EitherT.liftF(getTimestampForSnapshot(lastSnapshot.ordinal))
+
+      genesisBlockId = BlockIdentifier(genesisSnapshot.ordinal, Hex(genesisSnapshot.lastSnapshotHash.value))
+      currentBlockId = BlockIdentifier(lastSnapshot.ordinal, Hex(lastSnapshot.hash.value))
+    } yield
+      NetworkStatusResponse(
+        currentBlockIdentifier = currentBlockId,
+        currentBlockTimestamp = blockTimestamp,
+        genesisBlockIdentifier = genesisBlockId,
+        oldestBlockIdentifier = genesisBlockId,
+        syncStatus = SyncStatus(
```

Comment:
does it depend on the download process?

Reply:
🤔 I'm not sure. Do you think this should be `NodeState` ?
If so, for which states would _synced_ be `true`?

Reply:
I'm asking because I don't know what rosetta specifies under this property. anyway, imo it shouldn't be the node state (as it's custom to our network)

Reply:
but as I think about it, it says `stage = Synced` which is invalid if node has download in progress 🤔

Reply:
I haven't found anything yet that tells us what is expected. Since it's not mentioned in the schema I'm guessing it's our choice. Given that there's a `synced` flag I'm guessing `stage` is informative and that the flag is really telling whether the node is "ready".

Maybe we should have at least two values for staged, e.g. `Ready` when `synced = true` and `Not Ready`  otherwise.
Then we just need to decide which state (`NodeState` ?) will set the flag to true.

WDYT?
@ryle-goehausen any thoughts?

Reply:
We definitely don't need any more granularity other than Ready / not ready here, so synced = true when node state is Ready is fine.

This can also be optional and not required to implement as it's only for advanced uses.